main.tf

```diff
@@ -0,0 +1,82 @@
+
+variable "zone_id" {
+  description = "Zone id for record"
```

Comment:
Está faltando o type aqui.
Seria legal avaliar a possibilidade de colocar condition pra checar os valores também. Se for possível claro.

https://www.terraform.io/docs/language/values/variables.html#type-constraints
https://www.terraform.io/docs/language/values/variables.html#custom-validation-rules

Reply:
Adicionei o type=string.
Não vejo necessidade agora de colocar validation pq não temos regras de negocio para validar.

Na verdade temos uma regra que eu não adaptei, que é quando o registro é proxied o ttl tem que ser 1.
Porém essa validação é usando duas variaveis.
O que posso fazer é colocar um locals para fazer essa regra de negocio